lib/BlockedBy.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const NodeResque = require('node-resque');
+const { queuePrefix } = require('../config/redis');
+
+class BlockedBy extends NodeResque.Plugin {
+    async beforePerform() {
+        const { jobId, blockedBy } = await this.jobInfo();
+        const blockingJobKeys = await this.queueObject.connection.redis.mget(blockedBy);
+        const blockingJobsRunning = blockingJobKeys.some(j => j !== null);
+
+        // If any blocking job is running, then re-enqueue
+        if (blockingJobsRunning) {
+            await this.reEnqueue();
+
+            return false;
+        }
+
+        // Register the curent job as running by setting key
+        // Set expire time to take care of the case where
+        // afterPerform failed to call and blocked jobs will be stuck forever
+        await this.queueObject.connection.redis.expire(jobId, this.lockTimeout());
+
+        // Proceed
+        return true;
+    }
+
+    async afterPerform() {
+        const { jobId } = await this.jobInfo();
+
+        // Delete current job key
+        await this.queueObject.connection.redis.del(jobId);
+
+        return true;
+    }
+
+    async reEnqueue() {
+        await this.queueObject.enqueueIn(this.enqueueTimeout(), this.queue, this.func, this.args);
```

Comment:
How was `enqueueTimeout` chosen? Or is it an arbitrary low number just to act as a small buffer?

Reply:
It's only a random number. I'm not sure what number to set it as.
I'm not sure what the better name for it too. Maybe `waitTimeToReenqueue`?

Reply:
`reenqueueWaitTime`, maybe? Also, maybe `1800` to start the job again? Feel like waiting half an hour would be reasonable enough that we don't keep re-enqueuing it too much and at the same time don't wait too long. Not sure though, might be worth a discussion. (Another approach is to use the worker events to reenqueue the job once the blocking job is finished)